### PR TITLE
add human commiter to KG and clean up dataset creator ids

### LINKED ISSUE #632: `schema:creator` pointing to `mailto:None` for imported data-set
It looks like in cases when a data-set is imported, the `schema:creator` triple points to `<mailto:None>` resource. As a consequence, I imagine all the triples from all the projects and imported data-sets will be put into one `schema:Person` `<mailto:None>` bag during triples generation process.

Reproducing:
repo: `git@dev.renku.ch:virginiafriedrich/datasets-test.git`
commit: `46a77c18a31e42d356be7e6d4eac5ad35468e881`
command: `renku log --format rdf`

### LINKED ISSUE #634: No info about person on the commit triples
It looks that currently there's no information on the commit triples about a person who performed a commit. It would be great if that's added so we'd have something like follows:
```
  <rdf:Description rdf:about="file:///commit/<sha>">
    <prov:agent rdf:resource="file:///_<uuid>"/>
    ...
  </rdf:Description>
  ...
  <rdf:Description rdf:about="file:///_<uuid>">
    <rdf:type rdf:resource="http://schema.org/Person"/>
    <rdf:type rdf:resource="http://www.w3.org/ns/prov#Person"/>
    <schema:name>user name</schema:name>
    <schema:email><host@user></schema:email>
  </rdf:Description>
```

### PR DESCRIPTION
**Pull request by rokroskar**
This PR:

* prevents the creation of nodes with `mailto:None` ids - instead, if we don't have obvious identifying information like an email or an orcid-id we simply generate a blank node
* adds the `Person` `prov:agent` to the commit activity - this is the human agent that created the commit

This is a sample graph:

noname gv

Fixes #634 
Fixes #632

### CODE REVIEW

File: renku/models/datasets.py [hunk 1]
```diff
@@ -123,10 +123,12 @@ def from_commit(cls, commit):
...
 
-    def __attrs_post_init__(self):
-        """Post-Init hook to set _id field."""
-        if not self._id:
-            self._id = 'mailto:{self.email}'.format(self=self)
+    @_id.default
+    def default_id(self):
+        """Set the default id."""
+        if self.email:
+            return 'mailto:{self.email}'.format(self=self)
```
Comment (jsam) [inline renku/models/datasets.py]: I think we don't need to pass entire self context to format. I would just pass what's needed.

File: renku/models/datasets.py [hunk 2]
```diff
@@ -123,10 +123,12 @@ def from_commit(cls, commit):
...
+        return '_'+str(uuid.uuid4())
```
Comment (jsam) [inline renku/models/datasets.py]: Is this formatted? (yapf)

Comment (rokroskar) [reply renku/models/datasets.py]: nope! thanks!

File: renku/models/datasets.py (hunk 1 above)
Comment (rokroskar) [reply renku/models/datasets.py]: true - fixed

**Pull request merged**